fasm2bels/models/dsp_models.py

```diff
@@ -0,0 +1,449 @@
+                                    'BCIN[{}]'.format(idx))
+
+    bel.add_unconnected_port('BCOUT', 18, direction="output")
+    for idx in range(18):
+        bel.map_bel_pin_to_cell_pin(bel.bel, 'BCOUT{}'.format(idx),
+                                    'BCOUT[{}]'.format(idx))
+
+    bel.add_unconnected_port('CARRYCASCIN', None, direction="input")
+    bel.map_bel_pin_to_cell_pin(bel.bel, 'CARRYCASCIN', 'CARRYCASCIN')
+    bel.add_unconnected_port('CARRYCASCOUT', None, direction="output")
+    bel.map_bel_pin_to_cell_pin(bel.bel, 'CARRYCASCOUT', 'CARRYCASCOUT')
+
+    bel.add_unconnected_port('MULTSIGNIN', None, direction="input")
+    bel.map_bel_pin_to_cell_pin(bel.bel, 'MULTSIGNIN', 'MULTSIGNIN')
+    bel.add_unconnected_port('MULTSIGNOUT', None, direction="output")
+    bel.map_bel_pin_to_cell_pin(bel.bel, 'MULTSIGNOUT', 'MULTSIGNOUT')
+
+    bel.add_unconnected_port('PCIN', 48, direction="input")
+    for idx in range(48):
+        bel.map_bel_pin_to_cell_pin(bel.bel, 'PCIN{}'.format(idx),
+                                    'PCIN[{}]'.format(idx))
+
+    bel.add_unconnected_port('PCOUT', 48, direction="output")
+    for idx in range(48):
+        bel.map_bel_pin_to_cell_pin(bel.bel, 'PCOUT{}'.format(idx),
+                                    'PCOUT[{}]'.format(idx))
+
+    site.add_bel(bel, name='DSP48E1')
+    site.set_post_route_cleanup_function(cleanup_dsp(top, site))
+    top.add_site(site)
+
+def process_dsp(conn, top, tile, features):
+    """
+    Processes a DSP_[LR] tile with DSP48E1 sites.
+    
+    
+    DSP48 config:
+    
+    DSP48.DSP_[01].A_INPUT[0] 27_244
+    DSP48.DSP_[01].AREG_0 26_273 26_297 27_271
+    DSP48.DSP_[01].AREG_2 27_296
+    DSP48.DSP_[01].AUTORESET_PATDET_RESET 26_239
+    DSP48.DSP_[01].AUTORESET_PATDET_RESET_NOT_MATCH 26_238
+    DSP48.DSP_[01].B_INPUT[0] 26_171
+    DSP48.DSP_[01].BREG_0 26_200 26_208 27_198
+    DSP48.DSP_[01].BREG_2 27_207
+    DSP48.DSP_[01].MASK[47:0]
+    DSP48.DSP_[01].PATTERN[47:0]
+    DSP48.DSP_[01].SEL_MASK_C 26_243
+    DSP48.DSP_[01].SEL_MASK_ROUNDING_MODE1 27_242
+    DSP48.DSP_[01].SEL_MASK_ROUNDING_MODE2 27_241 27_242
+    DSP48.DSP_[01].USE_DPORT[0] 26_255
+    DSP48.DSP_[01].USE_SIMD_FOUR12 26_303 27_212
+    DSP48.DSP_[01].USE_SIMD_FOUR12_TWO24 26_244
+    DSP48.DSP_[01].ZADREG[0] 27_255
+    DSP48.DSP_[01].ZALUMODEREG[0] 26_214
+    DSP48.DSP_[01].ZAREG_2_ACASCREG_1 26_299
+    DSP48.DSP_[01].ZBREG_2_BCASCREG_1 27_209
+    DSP48.DSP_[01].ZCARRYINREG[0] 26_162
+    DSP48.DSP_[01].ZCARRYINSELREG[0] 27_170
+    DSP48.DSP_[01].ZCREG[0] 26_236
+    DSP48.DSP_[01].ZDREG[0] 27_253
+    DSP48.DSP_[01].ZINMODEREG[0] 26_247
+    DSP48.DSP_[01].ZIS_ALUMODE_INVERTED[3:0]
+    DSP48.DSP_[01].ZIS_CARRYIN_INVERTED 27_169
+    DSP48.DSP_[01].ZIS_CLK_INVERTED 27_237
+    DSP48.DSP_[01].ZIS_INMODE_INVERTED[4:0]
+    DSP48.DSP_[01].ZIS_OPMODE_INVERTED[6:0]
+    DSP48.DSP_[01].ZMREG[0] 26_198
+    DSP48.DSP_[01].ZOPMODEREG[0] 26_185
+    DSP48.DSP_[01].ZPREG[0] 27_235

```

Comment:
Please remove those from the docstring. This is important!  Those are bit mappings that should be distributed only with the prjxray-database.

Reply:
Do I need to remove all of it or just the bit mappings?

Reply:
Done.